src/lib/hooks/useCurrency.ts

```diff
@@ -150,9 +154,8 @@ type TokenMap = { [address: string]: Token }
  * Returns undefined if tokenAddress is invalid or token does not exist.
  */
 export function useTokenFromMapOrNetwork(tokens: TokenMap, tokenAddress?: string | null): Token | null | undefined {
-  const address = isAddress(tokenAddress)
+  const address = isAddress(tokenAddress?.toLowerCase())
```

Comment:
does this need to be lower-cased? seems like something that should happen within `isAddress`, if it is actually necessary.

Reply:
needs to be lowercase, added comment and moved it https://docs.ethers.io/v5/api/utils/address/